tests/foreman/sys/test_fam.py

```diff
@@ -54,10 +54,20 @@ def install_import_ansible_role(module_target_sat):
 
 
 @pytest.fixture(scope='module')
-def setup_fam(module_target_sat, module_sca_manifest, install_import_ansible_role):
+def setup_fam(
+    module_target_sat, module_sca_manifest, install_import_ansible_role, module_capsule_configured
+):
     # Execute AAP WF for FAM setup
     Broker().execute(workflow='fam-test-setup', source_vm=module_target_sat.name)
 
+    # Edit settings so FAM playbooks use the capsule we setup and then
+    # delete it from the Satellite so that the tests can use it.
+    settings.fam.server.foreman_proxy = module_capsule_configured.hostname
+    capsule = module_target_sat.api.SmartProxy().search(
+        query={'search': f'name = {module_capsule_configured.hostname}'}
+    )[0]
+    capsule.delete()
```

Comment:
Why does the capsule need to be deleted so tests can use it? Wouldn't it be better to use unconfigured capsule? (I know it would probably mean to create a new fixture)

Reply:
The `smart_proxy` and `katello_smart_proxy` modules upstream don't do any of the required setup for a capsule like generating certs or enabling repos or registration or installer, etc. That means we need a capsule that is pretty much ready to go on all fronts. However, you can only have 1 capsule per hostname:port. Since the test is using the hostname we pass for `foreman_proxy` and always uses port 9090 we need to make sure there are no capsules at `foreman_proxy:9090`. Also, I don't think there's any way to do all of the capsule setup without it actually being present in the list of capsules on the satellite, so the easiest way to do this IMO is to just do everything like normal then delete it from the list of capsules.